wit/streams.wit

```diff
@@ -259,4 +259,40 @@ interface streams {
             len: u64,
         ) -> result<u64, stream-error>;
     }
+
+    /// Continuously read all data from the input stream and write it to the
+    /// output stream.
+    ///
+    /// Forwarding is done on a background task. The tasks runs until either
+    /// stream closes or fails. After that, the streams are dropped and the
+    /// returned future is resolved. Dropping the future before it has been
+    /// resolved aborts the background task and drops the streams.
+    ///
+    /// The future resolves with the reason that caused the forwarding to terminate.
+    /// In case one of the streams ended normally, this will be `stream-error::closed`.
+    ///
+    /// This function requires exclusive access to the provided streams,
+    /// yet it does not change the ownership structure of them. I.e. they remain
+    /// children of their parent resources. If the parents place any lifetimes
+    /// restrictions on their children, those continue to apply.
+    forward: func(src: input-stream, dst: output-stream) -> future-forward-result;
+
+    /// Represents a future which may eventually return trailers, or an error.
```

Comment:
Since this is wasi-io and just focused on streams, it doesn't know anything about trailers per se.

If you want to mention how this interacts with trailers in wasi-http, I suggest putting that in a separate paragraph and introducing it with "In wasi-http, ...".

Reply:
Hah, no I don't want to mention HTTP here.
I copy-pasted the future type from wasi-http and apparently I forgot to edit this comment. :)